core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -3436,6 +3437,20 @@ object SparkContext extends Logging {
     supplement(DRIVER_JAVA_OPTIONS)
     supplement(EXECUTOR_JAVA_OPTIONS)
   }
+
+  private def supplementBlasOptions(conf: SparkConf): Unit = {
+    conf.getOption("spark.ml.allowNativeBlas").foreach { allowNativeBlas =>
+      def supplement(key: OptionalConfigEntry[String]): Unit = {
```

Comment:
This is repeated so many times now - I wonder if a simple refactor is in order to have one 'supplement' function?

Reply:
thanks for suggestion, will do

Reply:
refactored by extracting a common method `supplementJavaOpts`